lib/screens/authentication/import_key/interactor/viewmodels/import_key_bloc.dart

```diff
@@ -32,8 +34,20 @@ class ImportKeyBloc extends Bloc<ImportKeyEvent, ImportKeyState> {
       emit(state.copyWith(pageState: PageState.failure, error: ImportKeyError.InvalidPrivateKey));
     } else {
       final results = await ImportKeyUseCase().run(publicKey);
-      emit(ImportKeyStateMapper()
-          .mapResultsToState(state, results, AuthDataModel.fromKeyAndWords(event.privateKey, event.words)));
+      final List<Result> alternateResults = event.alternatePrivateKey != null
+          ? await ImportKeyUseCase().run(CheckPrivateKeyUseCase().isKeyValid(event.alternatePrivateKey!)!)
```

Comment:
CheckPrivateKeyUseCase().isKeyValid(event.alternatePrivateKey!) Can be null. 
adding `!` can lead to crash

Reply:
CheckPrivateKeyUseCase can be refactored to return Result. and avoid dealing with nulls.

Reply:
I know in this case the private key is always valid though.. will think about this

Reply:
Even though this can't happen, I refactored it so it now really can't happen.

Reply:
I agree that blank "!" are a no-no. Rather add a line of code and an additional check.